pkg/apis/openebs.io/v1alpha1/types.go

```diff
@@ -206,20 +206,25 @@ type CStorPoolList struct {
 // +resource:path=cstorvolumereplica
 // +genclient:nonNamespaced
 
-// CStorVolumeReplica describes a cstor pool resource created as custom resource
+// CStorVolumeReplica describes a cstor volume resource created as custom resource
 type CStorVolumeReplica struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              CStorVolumeReplicaSpec `json:"spec"`
+	Spec              CStorVolumeReplicaSpec   `json:"spec"`
+	Status            CStorVolumeReplicaStatus `json:"status"`
 }
 
 // CStorVolumeReplicaSpec is the spec for a CStorVolumeReplica resource
 type CStorVolumeReplicaSpec struct {
 	CStorControllerIP string `json:"cStorControllerIP"`
-	VolName           string `json:"volName"`
 	Capacity          string `json:"capacity"`
 }
 
+// CStorVolumeReplicaStatus is for handling status of cvr.
+type CStorVolumeReplicaStatus struct {
+	Phase string `json:"phase"` //init/online/offline/deletion-failed
```

Comment:
Please use enums to define various phases. Each of the phase should be commented with explanations about their significance & occurance

Reply:
As mentioned in above comment, the differrent status are listed and significance will be,
```
// Status written onto CStorPool and CStorVolumeReplica objects.
const (
	// StatusInit ensures the create operation is to be done, if import fails.
	StatusInit           = "init"
	// StatusOnline ensures the resource is available.
	StatusOnline         = "online"
	// StatusOffline ensures the resource is not available.
	StatusOffline        = "offline"
	// StatusDeletionFailed ensures the resource deletion has failed.
	StatusDeletionFailed = "deletion-failed"
	// StatusInvalid ensures invalid resource.
	StatusInvalid        = "invalid"
)
```
It will be present in cmd/cstor-pool-mgmt/controller/common, right now. Also, this file needs to be autogenerated and we are specifying many CRs, I feel we can keep the values of the business logic separate.

Reply:
@gkGaneshR please go ahead. typed string is better than enum.